packages/bpk-component-section-list/src/BpkSectionListItem.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import PropTypes from 'prop-types';
+import React, { type Node } from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkLargeChevronRightIcon from 'bpk-component-icon/lg/chevron-right';
+
+import STYLES from './bpk-section-list-item.scss';
+
+const getClassName = cssModules(STYLES);
+
+type Props = {
+  blank: boolean,
+  children: Node,
+  className: ?string,
+  href: ?string,
+  onClick: ?(event: SyntheticEvent<>) => void,
+};
+
+const BpkSectionListItem = (props: Props) => {
+  const { blank, children, className, href, onClick, ...rest } = props;
+  const classNames = [getClassName('bpk-section-list-item')];
+  if (className) {
+    classNames.push(className);
+  }
+
+  if (href) {
+    const target = blank ? '_blank' : null;
+    return (
+      <a
+        href={href}
+        target={target}
+        onClick={onClick}
+        className={classNames.join(' ')}
+        {...rest}
+      >
+        {children}
+        <BpkLargeChevronRightIcon
+          className={getClassName('bpk-section-list-item__chevron')}
+        />
+      </a>
+    );
+  }
+
+  if (onClick) {
+    return (
+      <button
+        type="button"
+        onClick={onClick}
+        className={classNames.join(' ')}
+        {...rest}
+      >
+        {children}
+        <BpkLargeChevronRightIcon
+          className={getClassName('bpk-section-list-item__chevron')}
+        />
+      </button>
+    );
+  }
+
+  return (
+    <div className={classNames.join(' ')} {...rest}>
+      {children}
+    </div>
+  );
+};
+
+BpkSectionListItem.propTypes = {
+  children: PropTypes.node.isRequired,
+  blank: PropTypes.bool,
+  className: PropTypes.string,
+  href: PropTypes.string,
+  onClick: PropTypes.func,
```

Comment:
The prop type of `href` and `onClick` could be stronger to ensure you don't supply both. Same applies for `blank` with `onClick`. Maybe we should have two components?

Reply:
> Maybe we should have two components?

We can definitely do that, however I based this off of `BpkLink` which does allow for both `href` and `onclick`: https://github.com/Skyscanner/backpack/blob/master/packages/bpk-component-link/src/BpkLink.js#L56, so worth bearing that in mind.

Reply:
Fair let's keep it as is then